src/main/resources/avro/alleleAnnotationmethods.avdl

```diff
@@ -0,0 +1,205 @@
+@namespace("org.ga4gh.methods")
+protocol AlleleAnnotationMethods {
+
+/*
+This protocol defines methods use to mine pre-calculated variant allele
+annotations.
+*/
+
+import idl "methods.avdl";
+import idl "variants.avdl";
+import idl "alleleAnnotations.avdl";
+import idl "metadata.avdl";
+/*
+variantannotations/search returns annotation for the alleles of Variants
+
+This allows the mining of allele-specific annotations on a VariantSet by
+either a region or by a set of genomic features. Where a region is supplied
+annotation of all alleles vs all features in the region is returned. Where a
+set of features is supplied, only annotations against these features (matching
+on featuretype and id) are returned and other overlapping features are ignored.
+
+variantannotationsets/search returns information on the input to the
+annotation. This will be a VariantSet and the reference data and software
+versions used in calculating the annotation.
+It is essential this information is exhaustive.
+*/
+
+/******************  /variantannotations/search  *********************/
+/** 
+This request maps to the body of `POST /variantannotations/search` as JSON
+*/
+record SearchVariantAnnotationsRequest {
+  /** Required. The ID of the variant annotation set to search over. */
+  string variantAnnotationSetId;
+
+  /**
+  Only return annotation for variants which have exactly this name
+  (case-sensitive, exact match).
+  */
+  union { null, string } name = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this
+  name. One of this field or `referenceId` or `features` is required.
+  (case-sensitive, exact match)
+  */
+  union { null, string } referenceName = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this
+  ID. One of this field or `referenceName` or `features` is required.
+  */
+  union { null, string } referenceId = null;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The beginning of the window (0-based, inclusive) for which variants with
```

Comment:
The group were keen to allow the convenience method of searching by featureIds, which is an alternative way to to specify a genomic region.